### PR TITLE
 Simplify camera calibration setup

### PR DESCRIPTION
**Pull request by g3force**
- generate lines and arcs from field parameters
- generate control points from camera index
- support more cameras than attached locally by specifying the global camera id and sending this instead in the detection frame
- generate rectangular penalty area for RoboCup 2018

### CODE REVIEW

File: src/shared/util/field_default_constants.h [hunk 1]
```diff
@@ -29,20 +29,16 @@
...
-const double kBoundaryWidth = 250.0;
-
-extern const std::size_t kNumFieldLines;
-extern const FieldLine kFieldLines[];
-
-extern const std::size_t kNumFieldArcs;
-extern const FieldCircularArc kFieldArcs[];
-
-extern const GVector::vector2d<double> kCameraControlPoints[4][4];
+namespace FieldConstantsRoboCup2018A {
```
Comment (mahi97) [inline src/shared/util/field_default_constants.h]: is this division A parameters ?
If it's division-A, field length, field width, and goal width should be change to new numbers.

File: src/shared/util/camera_calibration.cpp [hunk 2]
```diff
@@ -775,6 +759,70 @@ CameraParameters::AdditionalCalibrationInformation::
...
+    }
+
+    for (int i = 0; i < kNumControlPoints; ++i) {
+        control_point_field_xs[i]->setDouble(control_points[i].x);
+        control_point_field_ys[i]->setDouble(control_points[i].y);
+    }
+}
+
+std::vector<GVector::vector2d<double> > CameraParameters::AdditionalCalibrationInformation::
+generateCameraControlPoints(int cameraId, int numCamerasTotal, double fieldHeight, double fieldWidth) {
```
Comment (asctec-andre) [inline src/shared/util/camera_calibration.cpp]: This is a very helpful thing. It was never easy to get the control points right, even for experienced users.
Could you add images to the wiki which show the field and 1-8 cameras with their ID? I assume this helpful function will not work properly if one does not know which camera ID should be where.

File: src/shared/util/field_default_constants.h (hunk 1 above)
Comment (g3force) [reply src/shared/util/field_default_constants.h]: oh yes, sry. I missed that.
I've updated it with the correct dimensions for div A. The dimension of the center circle is not clear yet, but at least I moved it to the constants too.
Anyway, this are only the defaults when initially starting the vision.

File: src/shared/util/camera_calibration.cpp (hunk 2 above)
Comment (g3force) [reply src/shared/util/camera_calibration.cpp]: I added a picture to the wiki: https://github.com/RoboCup-SSL/ssl-vision/wiki/camera-calibration

I will also update the rest of the Wiki page later.

Comment (asctec-andre) [reply src/shared/util/camera_calibration.cpp]: Awesome, thanks!

Comment (g3force) [reply src/shared/util/camera_calibration.cpp]: Documented the rest as well now: https://github.com/RoboCup-SSL/ssl-vision/wiki/camera-calibration

**Comment by g3force** [comment]
Found a bug in the control point calculation and fixed it.

**Pull request merged**